### ISSUE TITLE
Packaging does not respect configuration property

### ISSUE THREAD

**Issue by ViktorHofer** [dotnet/corefx#34777]
@janvorli mentioned that during the packaging phase the passed in configuration property is not respected. My guess is that this only happens if neither Debug or Release is passed in. Jan tries to cross build corefx on x64 Linux targeting arm64 Linux.

`/home/janvorli/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19070.2/build/Packaging.targets(1255,5): error : Error when creating nuget packed package from /home/janvorli/git/corefx/artifacts/packages/Debug/specs/runtime.linux-arm64.Microsoft.Private.CoreFx.NETCoreApp.nuspec. System.IO.DirectoryNotFoundException: Could not find a part of the path '/home/janvorli/git/corefx/artifacts/bin/native/netcoreapp-Linux-Debug-arm64'. [/home/janvorli/git/corefx/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]`

Repro:
ROOTFS_DIR=/home/janvorli/git/coreclr/cross/rootfs/arm64 src/Native/build-native.sh arm64 cross Checked Linux outconfig netcoreapp-Linux-Checked-arm64 -portable
`./build.sh --configuration Checked /p:BuildNative=false /p:ArchGroup=arm64 `
Also it seems that our docs are still outdated and need to be adapted to the recent command changes.

cc @ericstj @wtgodbe

**Comment by ericstj** [comment]
This is not packaging related, its more about build plumbing for scripts / native and the conventions that we use for those to talk to each other.

`Checked` isn't a configuration string we expect/support at all.  This is documented [here](https://github.com/dotnet/corefx/blob/0823d020bf74f67e51e57422480d13265405f7ac/Documentation/coding-guidelines/project-guidelines.md).  I would imagine that the `--configuration` option is ignored by the managed build, even if it were allowed then I would imagine the managed build would choke on the "checked" string.  Since BuildNative=false was specified the packages failed to build because no native bits were present.  

Not sure why we're using "Checked" as the configuration string, but the logic for locating the native bits is here.
1. ArchGroup ConfigurationGroup and OSGroup are passed to the build-native script. https://github.com/dotnet/corefx/blob/0823d020bf74f67e51e57422480d13265405f7ac/src/Native/build-native.proj#L16

2. Scripts construct an output path based on those strings.
https://github.com/dotnet/corefx/blob/0823d020bf74f67e51e57422480d13265405f7ac/src/Native/build-native.sh#L367
https://github.com/dotnet/corefx/blob/0823d020bf74f67e51e57422480d13265405f7ac/src/Native/build-native.cmd#L104

3. Managed build expects to find those native bits at the same path:
https://github.com/dotnet/corefx/blob/1072799465f41315f2d041e09db17e0390445739/Directory.Build.props#L105-L106

I can imagine manually specifying that last property to force the build to use the checked output, I don't expect it would work by telling the managed build to use checked as described in OP.

**Comment by janvorli** [comment]
I am not sure what do you mean by "Checked isn't a configuration string we expect/support at all". We have always supported three build types - Debug, Release and Checked. From build-native.sh:
```
echo "BuildType can be: -debug, -checked, -release"
```

**Comment by ericstj** [comment]
That's the native build, but the managed/package build never understood "Checked" as a valid value for ConfigurationGroup and that's what is needed for the rest of the build to consume those binaries for both tests and packaging as I mentioned.  Here is the doc for what is valid to pass to the managed build: https://github.com/dotnet/corefx/blob/0823d020bf74f67e51e57422480d13265405f7ac/Documentation/coding-guidelines/project-guidelines.md#build-pivots.

In other words I don't think the following would have ever worked:
```
./build.sh --configuration Checked /p:BuildNative=false /p:ArchGroup=arm64 
```

But you might be able to do something like
```
./build.sh /p:BuildNative=false /p:ArchGroup=arm64 /p:NativeBinDir=/home/janvorli/git/corefx/artifacts/bin/native/netcoreapp-Linux-Checked-arm64
```

**Comment by janvorli** [comment]
Ah, I didn't know the corefx managed part had no concept of checked builds. In coreclr, it does.

**Comment by ViktorHofer** [comment]
@janvorli I'm going to close this. If you think we need support for checked builds in our managed part please reopen.
